src/services/transactions_queued.rs

```diff
@@ -0,0 +1,215 @@
+    let mut tx_iter = backend_transactions.results.into_iter();
+    // If we are not on the first page then we take the first item to get information on the previous page
+    let before_tx = if page_meta.offset == 0 {
+        None
+    } else {
+        tx_iter.next()
+    };
+    // Nonce of the first item in the next page (-1 if not present)
+    let edge_nonce = after_tx.map_or(-1, |tx| tx.nonce as i64);
+    // Nonce of the last item in the previous page (-1 if not present)
+    let previous_page_nonce = before_tx.map_or(-1, |tx| tx.nonce as i64);
+    let mut last_proccessed_nonce = previous_page_nonce;
+
+    let mut service_transactions: Vec<TransactionListItem> = Vec::new();
+    for (group_nonce, transaction_group) in
+        &tx_iter.group_by(|transaction| transaction.nonce as i64)
+    {
+        // Check if we need to add section headers
+        if last_proccessed_nonce < safe_nonce && group_nonce == safe_nonce {
+            // If the last nonce processed was the initial nonce (-1) and this group nonce is the current Safe nonce then we start the Next section
+            service_transactions.push(TransactionListItem::Label { label: Label::Next })
+        } else if last_proccessed_nonce <= safe_nonce && group_nonce > safe_nonce {
+            // If the last nonce processed was the initial nonce (-1) or the current Safe nonce and this group nonce higher than the Safe nonce then we start the Queue section
+            service_transactions.push(TransactionListItem::Label {
+                label: Label::Queued,
+            })
+        } // Else: If the last proccessed nonce is higher than the current Safe nonce then all headers should already be present
+
+        // Update last proccessed nonce
+        last_proccessed_nonce = group_nonce as i64;
+
+        // Make the group peekable for conflict type checks
+        let mut group_iter = transaction_group.peekable();
+        // There will be always at least one transaction for a group
+        let group_start_tx = group_iter.next().unwrap();
+        // Check if this group has the same nonce as the starting item of the next page
+        let is_edge_group = group_nonce == edge_nonce;
+        // This group has the same nonce as the last group from the previous page => group continues
+        let conflict_from_previous_page = previous_page_nonce == group_nonce;
+        // If there is more than 1 item in this group or we are in an edge group then we have a conflict
+        let has_conflicts = group_iter.peek().is_some() || is_edge_group;
+        // If we start a new conflict group then we should render the conflict header
+        if has_conflicts && !conflict_from_previous_page {
+            service_transactions.push(TransactionListItem::ConflictHeader {
+                nonce: group_nonce as u64,
+            })
+        }
+        // Add the one transaction that is always present
```

Comment:
This is the first one in the group. If this one doesn't exist `group_by` wouldn't have caught it. 🤔  What if we check `transaction_group.len() > 1` ? Would that simplify this?

Reply:
transaction_group and iterator doesn'T have a len property

Reply:
Also not sure what you simplify here :D

Reply:
hahaha I see, right iterator... Adding the conflict header ,maybe? But also not a lot of value would be added, I guess